modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/options/OptionsMediator.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.options;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+import static org.mule.runtime.core.api.util.ClassUtils.instantiateClass;
+import static org.mule.runtime.module.extension.internal.options.OptionsMediatorUtils.cloneAndEnrichMetadataKey;
+import static org.mule.runtime.module.extension.internal.options.OptionsMediatorUtils.getPartOrderMapping;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.api.meta.model.EnrichableModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterGroupModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterizedModel;
+import org.mule.runtime.api.options.Option;
+import org.mule.runtime.api.options.OptionBuilder;
+import org.mule.runtime.api.options.OptionResolvingException;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.extension.api.options.OptionsResolver;
+import org.mule.runtime.extension.internal.property.OptionPartModelProperty;
+import org.mule.runtime.module.extension.internal.loader.java.property.OptionsResolverFactoryModelProperty;
+import org.mule.runtime.module.extension.internal.runtime.ParameterValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.ValueResolvingException;
+import org.mule.runtime.module.extension.internal.util.IntrospectionUtils;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * //TODO
+ *
+ * @since 4.0
+ */
+public final class OptionsMediator<T extends ParameterizedModel & EnrichableModel> {
+
+  private T componentModel;
+  private MuleContext muleContext;
+  private Supplier<Object> NULL_SUPPLIER = () -> null;
+
+  /**
+   *
+   * @param componentModel
+   * @param muleContext
+   */
+  public OptionsMediator(T componentModel, MuleContext muleContext) {
+    this.componentModel = componentModel;
+    this.muleContext = muleContext;
+  }
+
+  /**
+   *
+   * @param parameterGroupName
+   * @param parameterValueResolver
+   * @return
+   */
+  public Set<Option> getGroupOptions(String parameterGroupName, ParameterValueResolver parameterValueResolver)
```

Comment:
group division still spins in my mind. Use the model name, and look for anything called that way.